starboard/build/config/BUILDCONFIG.gn

```diff
@@ -358,11 +362,34 @@ template("executable") {
 }
 
 template("shared_library") {
-  target_with_platform_configs(target_name) {
+  original_target_name = target_name
+  target_with_platform_configs(original_target_name) {
     target_type = "shared_library"
     forward_variables_from(invoker, "*", [ "install_target" ])
   }
 
+  if (build_with_separate_cobalt_toolchain) {
+    group("${target_name}_build") {
+      forward_variables_from(invoker, [ "testonly" ])
+      deps = [ ":${original_target_name}_loader($starboard_toolchain)" ]
+    }
+    if (current_toolchain == starboard_toolchain) {
+      executable("${original_target_name}_loader") {
+        forward_variables_from(invoker, [ "testonly" ])
+        sources = [ "//$starboard_path/starboard_loader.cc" ]
+
+        libs = [ "$root_build_dir/lib${original_target_name}.so" ]
```

Comment:
This needs to be discussed.

You can either have linking  via dlopen()/dlsym or  explicit linking as mentioned above. Can't do both things at the same time.

This will be discussed in go/modularcobalton1pp

Reply:
Is this fine to go in as is now and be followed up by you later?

Reply:
sgtm, unless @y4vor has any objections.

Reply:
fine by me too